src/main/java/org/skriptlang/skript/bukkit/recipes/elements/EffDiscoverRecipe.java

```diff
@@ -0,0 +1,90 @@
+package org.skriptlang.skript.bukkit.recipes.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.bukkitutil.NamespacedUtils;
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.Keyed;
+import org.bukkit.NamespacedKey;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.Recipe;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Set;
+
+@Name("Discover Recipe")
+@Description("Makes the specified players discover or forget the recipes.")
+@Examples({
+	"make player discover recipe \"my_recipe\"",
+	"make player undiscover recipe \"my_recipe\"",
+	"unlock recipe \"my_recipe\" for all players",
+	"lock recipe \"my_recipe\" for all players"
+})
+@Since("INSERT VERSION")
+public class EffDiscoverRecipe extends Effect {
+
+	static {
+		Skript.registerEffect(EffDiscoverRecipe.class,
+			"make %players% (discover|unlock) recipe[s] [with [the] (key|id)[s]] %strings%",
```

Comment:
is there a reason these syntax have recipe with key x built-in instead of `recipe with key %string%` being an expression itself?
also, this conflicts with the second pattern, so the second pattern won't work with variables.